plugins/woocommerce/src/RestApi/Routes/V4/Orders/Controller.php

```diff
@@ -0,0 +1,502 @@
+	}
+
+	/**
+	 * Get a single item.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_item( $request ) {
+		$order = wc_get_order( (int) $request['id'] );
+
+		if ( ! $this->is_valid_order_for_request( $order ) ) {
+			return $this->get_route_error_by_code( self::INVALID_ID );
+		}
+
+		return $this->prepare_item_for_response( $order, $request );
+	}
+
+	/**
+	 * Get collection of orders.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		/**
+		 * Filter the query arguments for a request.
+		 *
+		 * @param array           $args    Key value array of query var to query value.
+		 * @param WP_REST_Request $request The request used.
+		 * @since 10.2.0
+		 */
+		$query_args = apply_filters( $this->get_hook_prefix() . 'get_items_query', $this->query_utils->prepare_query( $request ), $request );
+		$results    = $this->query_utils->get_query_results( $query_args );
+		$items      = array();
+
+		foreach ( $results['results'] as $result ) {
+			$items[] = $this->prepare_response_for_collection( $this->prepare_item_for_response( $result, $request ) );
+		}
+
+		$pagination_util = new Pagination();
+		$response        = $pagination_util->add_headers( rest_ensure_response( $items ), $request, $results['total'], $results['pages'] );
+
+		return $response;
+	}
+
+	/**
+	 * Create a single item.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function create_item( $request ) {
+		if ( ! empty( $request['id'] ) ) {
+			/* translators: %s: post type */
+			return $this->get_route_error_by_code( self::RESOURCE_EXISTS );
+		}
+
+		try {
+			$order = new WC_Order();
+			$order->set_created_via( ! empty( $request['created_via'] ) ? sanitize_text_field( wp_unslash( $request['created_via'] ) ) : 'rest-api' );
+			$order->set_prices_include_tax( 'yes' === get_option( 'woocommerce_prices_include_tax' ) );
+
+			$this->update_utils->update_order_from_request( $order, $request, true );
+			$this->update_additional_fields_for_object( $order, $request );
+
+			/**
+			 * Fires after a single object is created via the REST API.
+			 *
+			 * @param WC_Order         $order    Inserted object.
+			 * @param WP_REST_Request $request   Request object.
+			 * @since 10.2.0
+			 */
+			do_action( $this->get_hook_prefix() . 'created', $order, $request );
+
+			$request->set_param( 'context', 'edit' );
+			$response = $this->prepare_item_for_response( $order, $request );
+			$response->set_status( WP_Http::CREATED );
+			$response->header( 'Location', rest_url( sprintf( '/%s/%s/%d', $this->namespace, $this->rest_base, $order->get_id() ) ) );
+
+			return $response;
+		} catch ( \WC_Data_Exception $e ) {
+			$data = $e->getErrorData();
+
+			if ( $order->get_id() ) {
```

Comment:
Should we check if `$order` exists here first?

Reply:
https://github.com/woocommerce/woocommerce/pull/60980/commits/c0c2019b4d3e48f370461180cd43be02ce0d7ea6